serviceregistration/consul/consul_service_registration.go

```diff
@@ -229,6 +232,22 @@ func (c *serviceRegistration) merge(conf map[string]string) error {
 		c.logger.Debug("config service_tags set", "service_tags", tags)
 	}
 
+  // Get user-defined meta tags to attach to the registered service name
+	metaTagsJSON, ok := conf["service_meta"]
+	var metaTags = map[string]string{}
+	if ok {
+		metaTagList := []map[string]string{}
+		err := json.Unmarshal([]byte(metaTagsJSON), &metaTagList)
+		if err != nil {
+			return nil, errors.New("service tags must be a dictionary of string keys and values")
+		}
+		metaTags = metaTagList[0]
```

Comment:
As mentioned elsewhere, I'm a little confused why we're unmarshaling this into a list first and then taking the first element. Could this be changed so that we don't expect a list?

Reply:
I don't recall why I made this choice in the context of the codebase I originally submitted this PR to, but agree it does not make any sense at all as in config, these values—as part of `storage` or `service_registration`— will be maps. Vaguely remember having trouble understanding if I should be using HCL blocks or objects.

I think my original intention was to allow something like the following config snippet

```hcl

storage "consul" {
  service_meta = {some_key = "some value"}
}
```

It's been a while, and I eventually ended up hacking around this shortcoming, so don't really have an informed opinion. Fixing.